### DISCUSSION TITLE
How do I change the default storage path of containerd for an rke2 cluster created by rancher?

### DISCUSSION THREAD

**Discussion by brain-wu** [rancher/rke2#4939] [Q&A]
rke2 版本v1.23.17-rke2r1

After I started the rke2 cluster with rancher, the default containerd data store is on the system disk, but my system disk is too small, I need to migrate it to the data directory on the data disk. I according to the official documentation in the/var/lib/rancher/rke2/agent/etc/containerd new created config/directory,config.toml.tmpl. file content is as follows:

disabled_plugins = []
imports = []
oom_score = 0
plugin_dir = ""
required_plugins = []
root = "/data/containerd"
state = "/data/k3s/containerd"
disabled_plugins = []
imports = []
temp = ""
version = 2

[cgroup]
  path = ""

[debug]
  address = ""
  format = ""
  gid = 0
  level = ""
  uid = 0

[grpc]
  address = "/data/k3s/containerd/containerd.sock"
  gid = 0
  max_recv_message_size = 16777216
  max_send_message_size = 16777216
  tcp_address = ""
  tcp_tls_ca = ""
  tcp_tls_cert = ""
  tcp_tls_key = ""
  uid = 0

[plugins."io.containerd.internal.v1.opt"]
  path = "/var/lib/rancher/rke2/agent/containerd"
[plugins."io.containerd.grpc.v1.cri"]
  stream_server_address = "127.0.0.1"
  stream_server_port = "10010"
  enable_selinux = false
  sandbox_image = "index.docker.io/rancher/pause:3.6"

[plugins."io.containerd.grpc.v1.cri".containerd]
  snapshotter = "overlayfs"
  disable_snapshot_annotations = true

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
          runtime_type = "io.containerd.runc.v2"
        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia]
          privileged_without_host_devices = false
          runtime_engine = ""
          runtime_root = ""

Then I restarted the rke2-service service, but the data was still on the system disk and my configuration did not take effect.

**Comment by brandond** [comment]
I can't quite tell what's going on with your config template, the formatting does not appear to have carried over well when you pasted it into your comment.

Have you considered just putting /var/lib/rancher on a dedicated partition?

**Comment by brain-wu** [comment]
The default rke2 cluster started by rancher is to deploy /var/lib/rancher under the root partition. My root partition space is small, how can I migrate it to the mount disk?

**Comment by brandond** [comment]
Move the files over, and remount the disk at /var/lib/rancher?

**Reply by brain-wu** [reply]
In addition to this, is there any configuration file that can be modified through configuration to take effect?

**Reply by brain-wu** [reply]
My data disk has been mounted and used, I cannot uninstall it and mount it again. Now I can only modify the configuration file, then change the path in the configuration file, and then move the data to the path I configured.

**Reply by brandond** [reply]
>  I according to the official documentation in the/var/lib/rancher/rke2/agent/etc/containerd new created config/directory,config.toml.tmpl. file content is as follows:

I'm not really following you here. Are you following the documentation at https://docs.rke2.io/advanced#configuring-containerd ? Can you confirm that you placed the file at `/var/lib/rancher/rke2/agent/etc/containerd/config.toml.tmpl` ?

Also note that v1.23 is long end of life, you should migrate to 1.26 or newer as soon as possible.
